rodeo/src/rodeo/06-either-and-try.scala

```diff
@@ -0,0 +1,199 @@
+
+    val names = List(
+      Success("David"),
+      Success("Vic"),
+      Success("Fabian"),
+      Failure(new IllegalArgumentException("Andrew")),
+      Failure(new IllegalArgumentException("Iván"))
+    )
+    val namesOffice = names.map {
+      case Failure(name) => "Lacayo rebelde"
+      case Success(name) => Success(name + "_" + "DealEngine")
+    }
+
+    assertTrue(namesOffice(1) == ???)
+  }
+
+  // We will now introduce the `Either[A,B]` type.
+  // An instance of the `Either[A,B]` type is either an instance of type A or an instance of type B. In mathematical terms `Either[A,B]` is the disjoint union of A and B, it is defined as a sum type.
+  // `Either` can be used in cases where a computation could return two different types depending on a certain condition imposed in the input.
+  // The `Either[A,B]` type is more general than the `Try[A]` type. Try[A] = Either[Failure[A],Success[A]]`.
+  // One difference in the case of exception handling between Try and Either is that by using Either we are not forced to represent failure by a wrapped instance of the Throwable class.
+  // We can understand `Either[A,B]` intuitively as a choice between left and right. For exception handling, left represents failure while right represents success.
+
+  Exercise("Tacos") {
+    def numberOfTacos(
+        weight: Double,
+        exercised: Boolean
+    ): Either[String, Double] =
+      if (weight < 0) Left("This is not a valid weight")
+      else Right(if (exercised) 2 * (weight / 20) else weight / 20)
+
+    assertTrue(numberOfTacos(80, false) == ???)
+  }
+
+  Exercise("SquareRootEither") {
+    def sqrt(x: Double): Either[String, Double] =
+      if (x < 0)
+        Left("Negative numbers don't have a square root in the real numbers")
+      else Right(Math.sqrt(x))
+
+    assertTrue(sqrt(4) == ???)
+    assertTrue(sqrt(-1) == ???)
+
+  }
+
+  // Either[A,B] type can be useful in other cases than exception handling.
+
+  Exercise("MixedDuplicator") {
+    def mixedDuplicator(x: Either[String, Double]): Either[String, Double] = {
+      x match {
+        case Left(x)  => Left(x * 2)
+        case Right(x) => Right(x * 2)
+      }
+    }
+    assertTrue(mixedDuplicator(Right(7)) == ???)
+    assertTrue(mixedDuplicator(Left("Hi")) == ???)
+  }
+
+  // We can also define functional methods in `Either[A,B]` like `map` and `flatMap`
+  // This functional methods are “right biased”. They will transform the Right case only and leave the Left case fixed.
+
+  Exercise("Map method on Either") {
+
+    assertTrue(Right(7).map((x: Int) => x * x) == ???)
+    assertTrue(Left("Peje").map((x: String) => x + x) == ???)
+  }
+
+  // flatMap is almost like map, but it allows us to chain operations on multiple Either[A, B].
+  
+  Exercise("FlatMap method on Either") {
+
+    case class SomeError()
+    def divInt(x: Int, y: Int): Either[SomeError, Int] = y match {
+      case 0 => Left(SomeError())
+      case _ => Math.floorMod(x, y) == 0 match {
```

Comment:
Perhaps add a small comment on the reason to use `floorMod` - something like: 'In order to guarantee that this division only returns Integer values, we take the modulus and floor it to make sure there is no reminder'

  (I understand why, but I am used to modulus/floor to simplify stuff)